src/main/scala/com/bazaarvoice/sswf/model/StepResults.scala

```diff
@@ -0,0 +1,91 @@
+package com.bazaarvoice.sswf.model
+
+sealed abstract class StepResult(message: Option[String]) {
+  def isSuccessful: Boolean
+  def isInProgress: Boolean
+}
+
+case class Success(message: Option[String]) extends StepResult(message) {
+  def this() = this(None)
```

Comment:
I see. I thought you didn't need to do this in Scala, since you have a companion object at your disposal.